Expensify/App
# Add withSecondaryInteractionHandler to handle secondary interactions

@roryabraham and Pullerbear friend, can you review this please?

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147468

### Tests
1. Applied this diff, since we currently have no place that uses secondary interactions [diff.txt](https://github.com/Expensify/ReactNativeChat/files/5655933/diff.txt)
1. Open an image in a modal
1. Long press the modal header
1. Verify `Secondary Action!` was logged
1. Desktop / Web: Right click the modal header
1. Verify `Secondary Action!` was logged, and that the context menu did not appear
1. Close the modal with the `X` button


### Screenshots
No UI changes
